haystack/components/generators/hugging_face_api.py

```diff
@@ -194,12 +201,17 @@ def run(self, prompt: str, generation_kwargs: Optional[Dict[str, Any]] = None):
         # update generation kwargs by merging with the default ones
         generation_kwargs = {**self.generation_kwargs, **(generation_kwargs or {})}
 
-        if self.streaming_callback:
-            return self._run_streaming(prompt, generation_kwargs)
+        # check if streaming_callback is passed
+        streaming_callback = streaming_callback or self.streaming_callback
+
+        if streaming_callback:
+            return self._run_streaming(prompt, streaming_callback, generation_kwargs)
 
         return self._run_non_streaming(prompt, generation_kwargs)
 
-    def _run_streaming(self, prompt: str, generation_kwargs: Dict[str, Any]):
+    def _run_streaming(
+        self, prompt: str, streaming_callback: Callable[[StreamingChunk], None], generation_kwargs: Dict[str, Any]
+    ):
```

Comment:
This implementation looks good but for consistency across generators I am wondering if we can update these methods to have similar `get_response` and `get_stream_response`. [(Reference here)](https://github.com/deepset-ai/haystack-core-integrations/blob/97094f7ecf9755d4128c6c7d09295ec8b91fa78d/integrations/google_ai/src/haystack_integrations/components/generators/google_ai/gemini.py#L209)
The code would look something like this. Wdyt? 
```suggestion
    res = self._client.text_generation(
            prompt, 
            details=True, 
            stream=streaming_callback is not None, 
            **generation_kwargs
        )

    replies = self._get_stream_response(res, streaming_callback) if streaming_callback else self._get_response(res)
    def get_stream_response(
        self, res_chunk: Iterable[TextGenerationStreamOutput], streaming_callback: Callable[[StreamingChunk], None], generation_kwargs: Dict[str, Any]
    ):
    chunk_metadata = {**asdict(token), **(asdict(chunk.details) if chunk.details else {})}
            stream_chunk = StreamingChunk(token.text, chunk_metadata)
            chunks.append(stream_chunk)
            streaming_callback(stream_chunk)  # type: ignore # streaming_callback is not None (verified in the run method)
```